File: server/models/Submission.js
const mongoose = require("mongoose");

const submissionSchema = new mongoose.Schema({
  problemId: mongoose.Schema.Types.ObjectId,
  language: String,
  code: String,
  status: String,
  executionTime: Number,
  memoryUsed: Number,
  results: [
    {
      testcaseNumber: Number,
      status: String,
      time: Number,
      memory: Number,
    },
  ],
  createdAt: {
    type: Date,
    default: Date.now,
  },
});

module.exports = mongoose.model("Submission", submissionSchema);

File: judge/src/main/java/com/codejudge/judge/model/Submission.java
package com.codejudge.judge.model;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;
import java.util.List;

@Document(collection = "submissions")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Submission {

    @Id
    private String id;

    private String problemId;

    private String username;

    private String language;   // python / cpp / java

    private String code;

    private String status;     // Accepted / WA / TLE / CE / RE

    private long executionTime;

    private long memoryUsed;

    private String compilationError;

    private List<SubmissionResult> results;

    private Instant createdAt;
}